src/main/client/_macros.ftl

```diff
@@ -226,16 +226,22 @@
         [#local result = result + [param.name]/]
       [#elseif language == "ts"]
         [#local convertedType = convertType(param.javaType, language)/]
-        [#local result = result + [param.name + (convertedType != "Object")?then(': ' + convertedType, '')]]
+        [#if param.optional!false]
+          [#local suffix = "?"/]
+        [#else]
+          [#local suffix = ""/]
+        [/#if]
+        [#local result = result + [param.name + suffix + (convertedType != "Object")?then(': ' + convertedType, '')]]
       [#elseif language == "go"]
         [#local convertedType = convertType(param.javaType, language)/]
         [#if api.method == "patch" && convertedType?ends_with("Request")]
           [#local convertedType = "map[string]interface{}"/]
         [/#if]
         [#local result = result + [convertValue(param.name, language) + (convertedType != "interface{}")?then(' ' + convertedType, ' interface{}')]]
       [#elseif language == "python"]
-        [#-- If the parameter is optional, give it a default value --]
-        [#if optional]
+        [#-- If the parameter is optional, give it a default value. Ideally we'd not use optional, which is driven by string comments, but
```

Comment:
pre-existing code uses this `optional` boolean, but if I combined these now, a lot more change ensue than I wanted to make